python/ray/data/dataset.py

```diff
@@ -2459,12 +2461,13 @@ def show(self, limit: int = 20) -> None:
     @ConsumptionAPI(
         if_more_than_read=True,
         datasource_metadata="row count",
-        pattern="Time complexity:",
+        pattern="Examples:",
     )
     def count(self) -> int:
-        """Count the number of records in the dataset.
-
-        Time complexity: O(dataset size / parallelism), O(1) for parquet
+        """Count the number of records in the dataset. For `Dataset`s
+        which only read Parquet files (created with :meth:`~ray.data.read_parquet`),
+        this method reads the file metadata to efficiently count the number of records
+        without reading in the entire data.
```

Comment:
Nit: docstring summary should be one line. Also, replace "records" with "rows" for consistency with our terminology elsewhere
```suggestion
        """Count the number of rows in the dataset. 
        
        For `Dataset`s
        which only read Parquet files (created with :meth:`~ray.data.read_parquet`),
        this method reads the file metadata to efficiently count the number of rows
        without reading in the entire data.
```

Reply:
We use the metadata for other APIs, too (`read_images`, `from_pandas`, etc.).

I'm wondering if we should just remove the thing about Parquet? It's an implementation detail

Reply:
i think the specific case around parquet has come up in questions from OSS users sometimes, which is why i expanded on the existing `O(1) for parquet` to clarify what it means. let me know if you think it's too confusing, and we can remove it instead

Reply:
Ah, got it. If people have been asking about, sounds good to keep.